blog/2018/12/21/a-little-christmas-template-cooking/index.markdown

```diff
@@ -0,0 +1,104 @@
+data:
+  bio: briandfoy
+  description:
+---
+The Advent Calendar has shown you many great ways to use Mojolicious, and since you already have Mojo installed you can use it for things besides web processing. Today's recipe uses The templating rendering engine for something other than web responses.
+---
+
+First, process some string templates. Here's an example lifted from the [Mojo::Template](https://mojolicious.org/perldoc/Mojo/Template), using the [squiggly heredoc syntax released in v5.26](https://www.effectiveperlprogramming.com/2016/12/strip-leading-spaces-from-here-docs-with-v5-26/)
+
+	use Mojo::Template;
+
+	my $mt = Mojo::Template->new;
+
+	say $mt->render(<<~'EOF');
+		% use Time::Piece;
+		<div>
+		  % my $now = localtime;
+		  Time: <%= $now->hms %>
+		</div>
+		EOF
+
+The lines with leading percent sings are Perl code. One of those lines loads a module, [Time::Piece](), and the other creates the variable `$now`. The `<%= %>` insert values inline. You can figure out the other template syntax on your own; it's all in the [module documentation](https://mojolicious.org/perldoc/Mojo/Template).
+
+You can invert that so that the source of the values comes from outside of the template. The `vars()` method turns on your ability to pass a hash to the template; the hash keys turn into values. Sometimes this is preferable to having too much logic in the presentation layer:
+
+	use v5.26;
+	use Mojo::Template;
+
+	my $mt = Mojo::Template->new->vars(1);
+
+	use Time::Piece;
+	my $now = localtime;
+
+	say $mt->render(<<~'EOF', { time => $now->hms } );
+		<div>
+		  Time: <%= $time %>
+		</div>
+		EOF
+
+It's just as easy to take that template from a file (or many files). This is the sort of thing I used to do this with [Template Toolkit](http://template-toolkit.org), a very fine and capable module that's as good as it ever was. But, I'm already using Mojo for quite a few things and it already has a templating engine. I can reduce my dependency count and focus on one templating language.
+
+Typically, Mojo templates use the extension _.ep_. Loop through all of the files that you specify on the command line and cook the ones that have that extension
+
+	use v5.14;
+	use Mojo::Template;
+
+	my $mt = Mojo::Template->new->vars(1);
+
+	use Time::Piece;
+	my $now = localtime;
+
+	foreach my $file ( @ARGV ) {
+		my $new_file = $file =~ s/\.ep\z//r;
+		open my $fh, '>:utf8', $new_file or do {
```

Comment:
There are a few opportunities to use Mojo::File and Mojo::Util here and the following example. Like spurt and encode rather than open with a layer and print to the filehandle. Entirely up to you.